lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -482,6 +484,126 @@ Function Nssm-Update-Dependencies
     }
 }
 
+Function Convert-HashToString
+{
+    <#
+    Convert a hash into a variable length string of KEY=VALUE seperated by spaces
+
+    Input:
+    {
+        key1: value1,
+        key2: value2
+    }
+
+    Output:
+    "key1=value1 key2=value2"
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [hashtable]$hash
+    )
+
+    return ($hash.GetEnumerator() | % { "$($_.Key)=$($_.Value)" }) -join " "
+}
+
+Function Convert-ArrayToHash
+{
+    <#
+    Converts an array with key=value elements into a hashtable
+
+    Input:
+    ["key1=value1", "key2=value2"]
+
+    Output:
+    {
+        key1: value1,
+        key2: value2
+    }
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [array]$array
+    )
+
+    $hash = @{}
+
+    foreach ($element in $array)
+    {
+        $hash = $hash + (ConvertFrom-StringData $element)
+    }
+    return $hash
+}
+
+Function Nssm-Update-Extra-Environment
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name,
+        [Parameter(Mandatory=$true)]
+        [hashtable]$env_vars
+    )
+
+    $env_vars_string = Convert-HashToString $env_vars
+
+    $cmd = "get ""$name"" AppEnvironmentExtra"
+    $results = Nssm-Invoke $cmd
+
+    if ($LastExitCode -ne 0)
+    {
+        $result.nssm_error_cmd = $cmd
+        $result.nssm_error_log = "$results"
+        Throw "Error updating environment variables for service ""$name"""
+    }
+
+    $current_env_vars = Convert-ArrayToHash $results
+
+    <# 
+    Update the appEnvironmentExtra values if:
+    - The services current environment variables does not match the desired environment variables.
+    - env_vars is not an empty hashtable
+    #>
+    If (((Convert-HashToString $env_vars) -ne (Convert-HashToString $current_env_vars)) -and ($env_vars.Count -gt 0)) 
+    {
+        $cmd = "set ""$name"" AppEnvironmentExtra ""$env_vars_string"""
+        $results = Nssm-Invoke $cmd
+
+        if ($LastExitCode -ne 0)
+        {
+            $result.nssm_error_cmd = $cmd
+            $result.nssm_error_log = "$results"
+            Throw "Error updating environment variables for service ""$name"""
+        }
+
+        $result.changed_by = "update-environment"
+        $result.previous_environment = $current_env_vars
```

Comment:
Let's try and not add more return values. This module is slated to be rewritten sometime soon and having more return values we potentially need to deprecate over time is not really wanted.